### DISCUSSION TITLE
How to collect the span that only exsits in a slow trace

### DISCUSSION THREAD

**Discussion by jackiepon31** [apache/skywalking#8207] [Q&A]
BackGround：
      In order to reduce the pression from es storage side and build a  swify way to  trouble shoot slow transaction.
I  wanna find  a way to  collect  the trace and its spans when the whole distributed trace time cost is above a certain value.
      But the spans-collector own by the client-side cannot  decide to collect the span or not when the whole trace is not end.

**Comment by wu-sheng** [comment]
I don't know what do you mean exactly, and what is `spans-collector`. There are not SkyWalking concept, and no span-oriented sampling mechanism even. Even, we keep segment in the storage, rather than separate span. So, can't follow your question.

**Reply by jackiepon31** [reply]
Thanks for your reply，
 Oh，sorry , I  will try to re-explain my point . 
Can  Skywalking keep the trace segment in storage only when the whole tracing time cost  is above a certain value. 
And why i need this, plz refer to the backgroud i 've offerd above

**Reply by wu-sheng** [reply]
For whole trace, no. This kind of thing could cause distributed global lock and time windows, which is really a bad idea for high performance system monitoring. We have tested and abandoned it back to 2015.
You could sample the slow segments of a trace, but it would not guarantee you could get the whole complete trace, because the fast segments will be ignored too.

**Reply by jackiepon31** [reply]
Ya , I got your point.  It's really tough right now , cause i did put myself in a contradictory place....
    If I try to sample the segments , the slow trace will occur even more uncommonly and I cannot troubleshoot the throttle of my systems under discomple trace.
    If I wanna catch and storage the slow trace dataset only,  i need a time-window based top-processing service, but surly we'll face the throttle from this service and we cannot explore the trace with high perfomance.

**Reply by wu-sheng** [reply]
Correct, this is the case.
